caluma/form/serializers.py

```diff
@@ -607,3 +617,32 @@ def update(self, instance, validated_data):
 
     class Meta(SaveAnswerSerializer.Meta):
         pass
+
+
+class SaveDocumentFileAnswerSerializer(SaveAnswerSerializer):
+    value = CharField(write_only=True, source="file")
+    value_id = PrimaryKeyRelatedField(read_only=True, source="file")
+
+    def set_file(self, validated_data):
+        file_name = validated_data.get("file")
+        file = models.File.objects.create(name=file_name)
+        validated_data["file"] = file
+        return validated_data
+
+    def create(self, validated_data):
+        validated_data = self.set_file(validated_data)
+        instance = super().create(validated_data)
+        return instance
+
+    def update(self, instance, validated_data):
+        if instance.file and instance.file.name is not validated_data["file"]:
+            instance.file.delete()
+            validated_data = self.set_file(validated_data)
+        elif not instance.file:
+            validated_data = self.set_file(validated_data)
+
+        instance = super().update(instance, validated_data)
+        return instance
```

Comment:
I'd prefer `return super().update(instance, validated_data)`
(also on L635)

Reply:
Thanks for noticing! This was a copy/paste mistake :)